krun/scheduler.py

```diff
@@ -443,12 +441,6 @@ def run(self):
         assert self.manifest.num_execs_left > 0
         self.platform.wait_for_temperature_sensors()
 
-        # If we get here, this is a real run, with a benchmark about to
-        # run. Results should never be in memory at this point (as they
-        # grow over time, and can get very large, thus potentially
-        # influencing the benchmark)
-        assert self.results is None
```

Comment:
It's kind of superseded by the `Results` change. No? I could bring it back if you still feel the same.